modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/fragment/renderer/FDSAdminFragmentRenderer.java

```diff
@@ -585,6 +585,13 @@ private JSONArray _getFiltersJSONArray(
 			(ObjectEntry objectEntry) -> {
 				Map<String, Object> properties = objectEntry.getProperties();
 
+				String fieldName = StringUtil.replace(
```

Comment:
As we are using the same TreeView for field selection we need to remove the specific notation used to traverse the API response and transform into the one accepted by oData filters.

It works the same for collections of objects like `universityStudents[]studentName` -> `universityStudents/studentName` as for direct object properties `universityStudents.universityName` -> `universityStudents/universityName`.

Probably there is a way to replace both `[]` and `.` in one shot.

Reply:
The FrontendDataSet component needs the dots & square brackets notation to parse the response https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/getLocalizedValue.ts.

We could think on a way to customize the TreeView to accept different separators based on the use case.

Reply:
```Java
fieldName.replaceAll("[\\[\\]\\.]", StringPool.FORWARD_SLASH);
```
This regexp replace every bracket by a slash, so we end up with `//`, the same occurs with the dot if we have more than one in the field. 

I've been tinkering with some regexp with no luck :shrug: .

Reply:
Oh right, you need to replace only `[]` and not all brackets? Then, something like
```java
fieldName.replaceAll("(\\[\\]|\\.)", StringPool.FORWARD_SLASH);
```

Reply:
> We could think on a way to customize the TreeView to accept different separators based on the use case

No, I don't think that would we wise. We want Clay components to be as vanilla as possible. I was thinking to modify `getLocalizedValue.ts`, and other related places, to have the same format as Clay. The goal would be to avoid transformation of formats we are doing in PR. I'm only asking how difficult would that be. If it is difficult, I'm fine with doing transformation like in the PR.

Reply:
> We want Clay components to be as vanilla as possible. 

Agree. I think that we were adding the separators in our wrapper component `FieldSelectModalContent` but we are doing it in the `getFields` utility.